feat(guides): Separate capitols about running Superface from testing guide (#36)
superfaceai/docs: docs/guides/run-capability.md

# Run Capability

> This guide describes how a capability can be used in any production Node.js application, without the use of Superface [remote registry](https://superface.ai/catalog).

Author: I removed talk about offline use in here since its not really described in guide and added tip about offline use in `Run the app` section. We can add link to offline use guide later.

main();
```

:::info

> Replace `scope/profile-name`, `UseCaseName` and inputs for `.perform` method with the use case details you actually want to use.

Author: Ok, I italized `Replace...` and left `For details...` in admonition. Can be?

For details on SuperfaceClient API, please consult [OneSDK reference](/reference/one-sdk-js).
:::

## Run the app

You can then run your app that performs the use case.

```shell
node app.js
```

:::tip Offline Use
To use capibility without the use of Superface [remote registry](https://superface.ai/catalog). You have to import capabilities into project and ensure `super.json` has valid paths to your capabilities.

TODO: link to offline use guide
:::

> ### DEBUG

Author: True, but it's not always API calls.

> ### DEBUG

Author: What about `Observe and debug OneSDK or Parser`

> ### DEBUG

Author: I'll omit the debug contexts and stay with title you suggested. 👍🏻

> Superface uses [DEBUG package](https://github.com/visionmedia/debug), which you can use when running or testing your app. To use it, you simply have to add environment variable `DEBUG="superface*"` before running tests.

Reviewer: Avoid simply and provide example.

````suggestion
OneSDK uses the [debug package](https://github.com/visionmedia/debug) which is useful for observing the behavior of the SDK and debugging. To use it, set environment variable to `DEBUG="superface*"` before running the application:

```shell
DEBUG="superface*" node app.js
```

````

There are mutiple parts of Superface, which implemented this package and created debug context:

in `OneSDK`:

- `superface:config`
- `superface:profile-parameter-validator`
- `superface:map-interpreter`
- `superface:http`, `superface:http:sensitive`
- `superface:superjson`
- `superface:failover`, `superface:failover:sensitive`
- `superface:bound-profile-provider`
- `superface:registry`
- `superface:usecase`
- `superface:lib/env`
- `superface:events`
- `superface:metric-reporter`

in `Parser`:

- `superface-parser:map-validator`

> - `superface-parser:profile-io-analyzer`

Reviewer: I think this isn't much useful without further explanation _why_ I should use these contexts. I would move it to reference/README for OneSDK and link it here.

However, I would explicitly mention `superface:http*` which is extremely useful for debugging API calls, perhaps with a note about `:sensitive` context.

To know more about setting up specific contexts, check out [conventions](https://github.com/visionmedia/debug#conventions) and [wildcards](https://github.com/visionmedia/debug#wildcards) of debug package.

> > Note: Later you'll be able to publish your capability to the Superface registry. This will give you many advantages over usage from the local files, like: injecting latest map, provider failover, observability dashboard, any more. Stay tuned.

Reviewer: I would drop this paragraph.

```suggestion
```

main();
```

_Replace `scope/profile-name`, `UseCaseName` and inputs for `.perform` method with the use case details you actually want to use._

:::info
For details on SuperfaceClient API, please consult [OneSDK reference](/reference/one-sdk-js).
:::

## Run the app

Now run the application to perform the use case and check the results:

```shell
node app.js
```

:::tip Offline Use
To use capability without the use of Superface [remote registry](https://superface.ai/catalog). You have to import capabilities into project and ensure `super.json` has valid paths to your capabilities.

<!-- TODO: link to offline use guide -->
:::

### Observe and debug API calls

OneSDK uses the [debug package](https://github.com/visionmedia/debug) which is useful for observing the behavior of the SDK and debugging. To use it, set environment variable to `DEBUG="superface*"` before running the application:

```shell
DEBUG="superface*" node app.js
```

> To observe API calls, you can use `superface:http` debug context.

Reviewer: ````suggestion
To observe just the API calls, you can use `superface:http*` debug context:

```shell
DEBUG="superface:http*" node app.js
```

````

:::info Credentials
If you want to debug sensitive information, you can use context `superface:http:sensitive`

> :::

Reviewer: I recommend tuning this admonition around like: by default with * this includes sensitive data.

```
:::caution Sensitive Output

Using the `superface:http*` context will output full HTTP requests and responses including API credentials and other sensitive data. 

:::
```

:::note Other debug contexts
There are mutiple parts of Superface, which implemented this package and created debug context, you can find more about these contexts in [OneSDK repository](https://github.com/superfaceai/one-sdk-js#usage) or [Parser repository](https://github.com/superfaceai/parser) on Github.

> :::

Reviewer: This isn't documented yet, I suggest commenting this out for now.
